src/sample.jl

```diff
@@ -434,8 +445,10 @@ function mcmcsample(
                         progress && put!(channel, true)
 
                         # Return the new chain.
-                        return chain
+                        return chain, rng
                     end
+                    chains = first.(chains_rngs)
+                    rngs .= last.(chains_rngs)
```

Comment:
This seems to be an unnecessary change?
```suggestion
                        return chain
                    end
```